components/timezone/src/iana_ids.rs

```diff
@@ -0,0 +1,243 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::error::TimeZoneError;
+use crate::provider::names::*;
+use crate::TimeZoneBcp47Id;
+use icu_provider::prelude::*;
+
+/// A mapper from IANA time zone identifiers to BCP-47 time zone identifiers.
+///
+/// # Examples
+///
+/// ```
+/// use icu::timezone::IanaToBcp47Mapper;
+///
+/// let mapper = IanaToBcp47Mapper::new();
+/// let mapper_borrowed = mapper.as_borrowed();
+///
+/// // The IANA zone "Australia/Melbourne" is the BCP-47 zone "aumel"
+/// assert_eq!(
+///     mapper_borrowed.get("Australia/Melbourne"),
+///     Some("aumel".parse().unwrap())
+/// );
+///
+/// // Lookup is ASCII-case insensitive
+/// assert_eq!(
+///     mapper_borrowed.get("australia/melbourne"),
+///     Some("aumel".parse().unwrap())
+/// );
+///
+/// // The IANA zone "Australia/Victoria" is an alias
+/// assert_eq!(
+///     mapper_borrowed.get("Australia/Victoria"),
+///     Some("aumel".parse().unwrap())
+/// );
+/// ```
+#[derive(Debug)]
+pub struct IanaToBcp47Mapper {
+    data: DataPayload<IanaToBcp47MapV1Marker>,
+}
+
+impl IanaToBcp47Mapper {
+    /// Creates a new [`IanaToBcp47Mapper`].
+    ///
+    /// See [`IanaToBcp47Mapper`] for an example.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    #[inline]
+    pub const fn new() -> Self {
+        IanaToBcp47Mapper {
+            data: DataPayload::from_static_ref(
+                crate::provider::Baked::SINGLETON_TIME_ZONE_IANA_TO_BCP47_V1,
+            ),
+        }
+    }
+
+    icu_provider::gen_any_buffer_data_constructors!(locale: skip, options: skip, error: TimeZoneError,
+        #[cfg(skip)]
+        functions: [
+            new,
+            try_new_with_any_provider,
+            try_new_with_buffer_provider,
+            try_new_unstable,
+            Self,
+        ]
+    );
+
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::new)]
+    pub fn try_new_unstable<P>(provider: &P) -> Result<Self, TimeZoneError>
+    where
+        P: DataProvider<IanaToBcp47MapV1Marker> + ?Sized,
+    {
+        let data = provider.load(Default::default())?.take_payload()?;
+        Ok(Self { data })
+    }
+
+    /// Returns a borrowed version of the mapper that can be queried.
+    ///
+    /// This avoids a small potential cost of reading the data pointer.
+    pub fn as_borrowed(&self) -> IanaToBcp47MapperBorrowed {
```

Comment:
The borrowed type complicates the API, have you benchmarked if this makes a difference?

Reply:
No but I'm following the design we've established several places elsewhere for structures that are basically data lookup tables